crash when adding current position WP to not offline stored caches

**Describe the bug:**
crash when adding current position as WP to an not saved geocache

**To Reproduce:**
1. Search for GC40N20 or other GC number (w or w/o waypoints)
2. Swipe left to WP page
3. Click on Add current position

**Actual behavior/state after performing these steps:**
c:geo closes

**Expected behavior/state after performing these steps:**
WP with current postion is created

**Version of c:geo used:**
2020.11.30-NB-5108dbf

**Is the problem reproducible:**
Yes

**Additional context:**
If cache is stored, no crash, but WP is created.
Logfile sent to support

I took out the crash part only for now:

```
--------- beginning of crash
11-19 23:40:23.644  9488 10291 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
11-19 23:40:23.644  9488 10291 E AndroidRuntime: Process: cgeo.geocaching, PID: 9488
11-19 23:40:23.644  9488 10291 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at android.os.AsyncTask$4.done(AsyncTask.java:399)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:919)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: Caused by: java.lang.RuntimeException: Can't create handler inside thread Thread[AsyncTask #1,5,main] that has not called Looper.prepare()
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at android.os.Handler.<init>(Handler.java:207)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at android.os.Handler.<init>(Handler.java:119)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.utils.SimpleHandler.<init>(SimpleHandler.java:16)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$ChangeNotificationHandler.<init>(CacheDetailActivity.java:2548)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged(CacheDetailActivity.java:886)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.ensureSaved(CacheDetailActivity.java:1859)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.access$4500(CacheDetailActivity.java:183)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator$3.doInBackground(CacheDetailActivity.java:1951)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator$3.doInBackground(CacheDetailActivity.java:1947)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-19 23:40:23.644  9488 10291 E AndroidRuntime: 	... 4 more
11-19 23:40:33.234 12171 12247 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
11-19 23:40:33.234 12171 12247 E AndroidRuntime: Process: cgeo.geocaching, PID: 12171
11-19 23:40:33.234 12171 12247 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at android.os.AsyncTask$4.done(AsyncTask.java:399)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:919)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: Caused by: java.lang.RuntimeException: Can't create handler inside thread Thread[AsyncTask #1,5,main] that has not called Looper.prepare()
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at android.os.Handler.<init>(Handler.java:207)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at android.os.Handler.<init>(Handler.java:119)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.utils.SimpleHandler.<init>(SimpleHandler.java:16)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$ChangeNotificationHandler.<init>(CacheDetailActivity.java:2548)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.notifyDataSetChanged(CacheDetailActivity.java:886)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.ensureSaved(CacheDetailActivity.java:1859)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity.access$4500(CacheDetailActivity.java:183)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator$3.doInBackground(CacheDetailActivity.java:1951)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at cgeo.geocaching.CacheDetailActivity$WaypointsViewCreator$3.doInBackground(CacheDetailActivity.java:1947)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at android.os.AsyncTask$3.call(AsyncTask.java:378)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
11-19 23:40:33.234 12171 12247 E AndroidRuntime: 	... 4 more
```

--- a/main/src/cgeo/geocaching/CacheDetailActivity.java
+++ b/main/src/cgeo/geocaching/CacheDetailActivity.java
@@ -1918,26 +1918,18 @@ public class CacheDetailActivity extends AbstractViewPagerActivity<CacheDetailAc
             view.setOnScrollListener(new FastScrollListener(view));
 
             final Button addWaypointCurrent = header.findViewById(R.id.add_waypoint_currentlocation);
-            addWaypointCurrent.setOnClickListener(v -> new AsyncTask<Void, Void, Boolean>() {
-                @Override
-                protected Boolean doInBackground(final Void... params) {
-                    ensureSaved();
-                    final Waypoint newWaypoint = new Waypoint(Waypoint.getDefaultWaypointName(cache, WaypointType.WAYPOINT), WaypointType.WAYPOINT, true);
-                    newWaypoint.setCoords(Sensors.getInstance().currentGeo().getCoords());
-                    newWaypoint.setGeocode(cache.getGeocode());
-                    cache.addOrChangeWaypoint(newWaypoint, true);
+            addWaypointCurrent.setOnClickListener(v -> {
+                ensureSaved();
+                final Waypoint newWaypoint = new Waypoint(Waypoint.getDefaultWaypointName(cache, WaypointType.WAYPOINT), WaypointType.WAYPOINT, true);
+                newWaypoint.setCoords(Sensors.getInstance().currentGeo().getCoords());
+                newWaypoint.setGeocode(cache.getGeocode());
+                if (cache.addOrChangeWaypoint(newWaypoint, true)) {
                     addWaypointAndSort(sortedWaypoints, newWaypoint);
-                    return true;
+                    adapter.notifyDataSetChanged();
+                    reinitializePage(Page.WAYPOINTS);
+                    ActivityMixin.showShortToast(CacheDetailActivity.this, getString(R.string.waypoint_added));
                 }
-
-                @Override
-                protected void onPostExecute(final Boolean result) {
-                    if (result) {
-                        adapter.notifyDataSetChanged();
-                        ActivityMixin.showShortToast(CacheDetailActivity.this, getString(R.string.waypoint_added));
-                    }
-                }
-            }.execute());
+            });
 
             // read waypoint from clipboard
             final Button addWaypointFromClipboard = header.findViewById(R.id.add_waypoint_fromclipboard);

